django/db/models/base.py

```diff
@@ -1290,10 +1290,32 @@ def check(cls, **kwargs):
                 *cls._check_indexes(databases),
                 *cls._check_ordering(),
                 *cls._check_constraints(databases),
+                *cls._check_default_pk(),
             ]
 
         return errors
 
+    @classmethod
+    def _check_default_pk(cls):
+        if (
+            cls._meta.pk.auto_created and
+            not settings.is_overridden('DEFAULT_AUTO_FIELD') and
+            not cls._meta.app_config._is_default_auto_field_overridden
+        ):
+            return [
+                checks.Error(
+                    'Auto-created primary key used without defining a primary key type.',
+                    hint=(
+                        'Configure the DEFAULT_AUTO_FIELD setting or the %s.default_auto_field '
+                        'attribute to point to a subclass of AutoField, e.g. '
+                        "'django.db.models.BigAutoField'."
+                    ) % cls._meta.app_config.__class__.__name__,
+                    obj=cls,
+                    id='fields.E101',
```

Comment:
IMO we shouldn't raise an `Error` here, `Warning` sound fine. We don't really want to force all users to `DEFAULT_AUTO_FIELD` to their settings. Moreover we will probably remove this check after changing the default. I will push edits.